### DISCUSSION TITLE
sagaMiddleware concatenation error

### DISCUSSION THREAD

**Discussion by 0Ieg** [reduxjs/redux-toolkit#4053] [Q&A]
Guys, you still haven't fixed the sagaMiddleware concatenation error.
Look, an error appears in my [GTA-6 repository](https://github.com/0Ieg/GTA-6). See the latest commit 'Toolkit and saga error'
https://github.com/0Ieg/GTA-6

![toolkit and saga](https://github.com/reduxjs/redux-toolkit/assets/111610358/f269fd18-780e-4323-9287-83e9aa20f31a)

**I did everything according to your example**

![pr](https://github.com/reduxjs/redux-toolkit/assets/111610358/0c6a7f83-4ecb-4c7c-bff1-46c5b0a26943)

**Comment by markerikson** [comment]
First, you haven't posted anything previously that I know of about this TS error.

Second, that's a very demanding way to complain about things, and not helpful.

Third, that example is from the `redux-saga` docs, and not the RTK docs.

Fourth, this is not an issue with RTK. This is a problem with `redux-saga` needing an updated release that works with the RTK 2.0 TS types.

We actually filed an issue with `redux-saga` back in November to notify them that RTK 2.0 (and Redux 5.0) would have updated TS types, and that they'd need to update their TS types accordingly:

- https://github.com/redux-saga/redux-saga/issues/2401#event-11367989681

The saga maintainers have made changes, and that _should_ be fixed in `redux-saga@1.3.0`:

- https://github.com/redux-saga/redux-saga/releases/tag/redux-saga%401.3.0

I see that you have `"redux-saga": "^1.2.3"` listed, so in theory you should be getting v1.3.0.  However, it doesn't look like _you_ have updated:

```js
 "node_modules/redux-saga": {
      "version": "1.2.3",


 "node_modules/@redux-saga/core": {
      "version": "1.2.3",
```

Please update to `redux-saga@1.3.0` and that should fix this issue.

**Reply by 0Ieg** [reply]
Thanks, I get it. I have no complaints.

https://github.com/reduxjs/redux-toolkit/discussions/3982

**Reply by 0Ieg** [reply]
They updated the saga only 7 days ago. Now I have installed a new one, everything is OK
